File: src/lib/household-roles.ts
export type HouseholdRole = "owner" | "parent" | "guest";

export function canManageHousehold(role: HouseholdRole) {
  return role === "owner" || role === "parent";
}

export function isGuest(role: HouseholdRole) {
  return role === "guest";
}

export function roleLabel(role: HouseholdRole) {
  if (role === "owner") return "Owner";
  if (role === "parent") return "Parent";
  return "Guest";
}

File: ios/HomeHub/Utilities/HouseholdRoles.swift
import Foundation

enum HouseholdRoles {
    static func canManageHousehold(role: HouseholdRole) -> Bool {
        role == .owner || role == .parent
    }

    static func isGuest(role: HouseholdRole) -> Bool {
        role == .guest
    }

    static func roleLabel(_ role: HouseholdRole) -> String {
        switch role {
        case .owner: "Owner"
        case .parent: "Parent"
        case .guest: "Guest"
        }
    }
}
